opentelemetry/src/sdk/trace/span.rs

```diff
@@ -173,34 +173,49 @@ impl crate::trace::Span for Span {
 impl Span {
     fn ensure_ended_and_exported(&mut self, timestamp: Option<SystemTime>) {
         if let Some(mut data) = self.data.take() {
```

Comment:
I think while you're in here it would be good to further reduce rightward drift. You can turn the `if let Some(mut data) = self.data.take()` and `if let Some(timestamp) = timestamp` into `let foo = match bar { Some(x) => x, None => return };` expressions, which would really help.

Reply:
👍